superset/charts/data/api.py

```diff
@@ -327,12 +327,7 @@ def _run_async(
         """
         Execute command as an async query.
         """
-        # First, look for the chart query results in the cache.
-        with contextlib.suppress(ChartDataCacheLoadError):
-            result = command.run(force_cached=True)
```

Comment:
I believe that this line should refine and the bug lies in this line. This part of codes want to imminently response a *cached results* so do you mind to find a way to check if the cache key in the cache? We have [has](https://github.com/apache/superset/blob/a4c771e013957e95d80d252dcdedad7046348964/superset/common/utils/query_cache_manager.py#L214) method in QueryCacheManager that might help to this function.

Reply:
@zhaoyongjie ,
not sure understand what you meant

Here an example:
`    def _run_async(
        self, form_data: dict[str, Any], command: ChartDataCommand
    ) -> Response:
        """
        Execute command as an async query.
        """
        # First, look for the chart query results in the cache.
        with contextlib.suppress(ChartDataCacheLoadError):
            # Check if the cache key exists in the cache
            if QueryCacheManager.has(form_data.get("cache_key"), region=CacheRegion.DEFAULT):
                result = command.run(force_cached=True)
                if result is not None:
                    return self._send_chart_response(result)

        # Otherwise, kick off a background job to run the chart query.`
        
       Did not get where from I am supposed to get cache_key and region parameters to pass them to QueryCacheManager

Reply:
Did not get where from I am supposed to get cache_key and region parameters to pass them to QueryCacheManager

Reply:
@GAlexeyV will check the details codes and reply you tomorrow.

Reply:
@zhaoyongjie , thank you
I will finish it then when return back from holidays (in 2 weeks)

Reply:
Enjoy your vacation! I'll check the codes and then reply you.

Reply:
@zhaoyongjie hi, 
Did you have a chance to look into?

Reply:
I'm also looking ahead for this change to get merged.

Reply:
@john-bodley hi, maybe you can have a look?